Pass env vars in "KG_WHITELIST" to kernel endpoint

ricedavida opened the pull request:
This feature is designed to be paired with Jupyter Kernel Gateways env_whitelist option.
"--JupyterWebsocketPersonality.env_whitelist=<List>
    Default: []
    Environment variables allowed to be set when a client requests a new kernel"

To use this feature, simply add the env var names into "KG_WHITELIST" using a colon delimited list.

example:
export KG_WHITELIST=SPARK_OPTS:SPARK_USER

ricedavida committed:
Pass env vars in "KG_WHITELIST" to kernel endpoint

This feature is designed to be paired with Jupyter Kernel Gateways env_whitelist option.
"--JupyterWebsocketPersonality.env_whitelist=<List>
    Default: []
    Environment variables allowed to be set when a client requests a new kernel"

To use this feature, simply add the env var names into "KG_WHITELIST" using a colon delimited list.

example:
export KG_WHITELIST=SPARK_OPTS:SPARK_USER

parente commented:
/cc @jtyberg

jtyberg commented on nb2kg/nb2kg/managers.py:
```diff
@@ -113,7 +113,7 @@ def start_kernel(self, kernel_id=None, path=None, **kwargs):
                 body=json_encode({
                     'name': kernel_name,
-                    'env': {k:v for (k,v) in dict(os.environ).items() if k.startswith('KERNEL_')}
+                    'env': {k:v for (k,v) in dict(os.environ).items() if k.startswith('KERNEL_') or k in os.environ['KG_WHITELIST'].split(":")}
```
I would propose changes to maintain consistency with KG:

* [KG uses `KG_ENV_WHITELIST`](https://github.com/jupyter/kernel_gateway/blob/595e34ababe24f88697968c7deb7de735760a6ae/kernel_gateway/jupyter_websocket/__init__.py#L33), so use that name instead of using a new name `KG_WHITELIST` for the same variable.
* [KG uses comma `,` as delimiter](https://github.com/jupyter/kernel_gateway/blob/595e34ababe24f88697968c7deb7de735760a6ae/kernel_gateway/jupyter_websocket/__init__.py#L40), not `:`.

Other than that, looks fine.

ricedavida committed:
Refactoring KG_WHITELIST to KG_ENV_WHITELIST

To better align with Jupyter Kernel Gateway, this commit will change from
KG_WHITELIST to KG_ENV_WHITELIST and use comma as a delimeter instead of a colon.

ricedavida commented:
@jtyberg I updated the Pull Request to align with your suggestions.  It is now KG_ENV_WHITELIST and has a comma delimited list.
